portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,40 @@ const randomFactGenerator = () => {
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText = randomFact;
 };
+
+/**
+ * Fetches a the content from the server and adds it to the DOM.
+ */
+const getServletContent = () => {
+    console.log('Fetching the content from the server.');
+
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/data');
+
+    // When the request is complete, pass the response into handleResponse().
```

Comment:
Maybe we don't need these comments, since the code is so nice and simple. Also, could be

```
const f = () => {
  fetch('/data').then(handleResponse);
};
```

Which is to say that you don't need to store `responsePromise` or `log()` what's going on (that'll show up in the Chrome devtools).

Reply:
Yeah, I really was following the examples they gave us here, but later on in the following steps, I've been using a simple arrow function to do everything.